src/components/sync-dialog/SyncDialog.jsx

```diff
@@ -41,20 +40,25 @@ class SyncDialog extends React.Component {
     };
 
     handleSynchronization = async () => {
-        this.props.loading.show(true, i18n.t("Synchronizing metadata"));
+        const { handleClose, loading, metadata, d2 } = this.props;
+        loading.show(true, i18n.t("Synchronizing metadata"));
 
         try {
-            const syncReport = await startSynchronization(this.props.d2, {
-                metadata: this.props.metadata,
+            const builder = {
+                metadata: metadata,
                 targetInstances: this.state.targetInstances,
```

Comment:
[Very minor] We could unpack `targetInstances` from state at the top of the method, that way we see in a glance what props/state this method deals with.

Reply:
Sure!